### PR TITLE
fix: Fixed top error handling on server's request handler to ensure proper error boundary.

### PR DESCRIPTION
**Pull request by christerswahn**
Fixed top error handling on server's request handler to ensure proper error boundary and returning 500 error code.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

n/a

### CODE REVIEW

File: packages/serverpod/lib/src/server/server.dart [hunk 1]
```diff
@@ -184,8 +176,24 @@ class Server {
     stdout.writeln('$name stopped');
   }
 
+  void _handleRequestWithErrorBoundary(HttpRequest request) async {
+    // [Future.sync] ensures no synchronous error is accidentally thrown from the handler
+    // https://dart.dev/libraries/async/futures-error-handling#solution-using-future-sync-to-wrap-your-code
+    await Future.sync(() {
+      return _handleRequest(request);
```
Comment (SandPod) [inline packages/serverpod/lib/src/server/server.dart:183]: Question: Is no await required here?

**Review by SandPod — COMMENTED** [review]
Good catch on this!

The change looks good to me.

File: packages/serverpod/lib/src/server/server.dart (hunk 1 above)
Comment (christerswahn) [reply packages/serverpod/lib/src/server/server.dart:183]: The callback returns a FutureOr which is passed on to the caller if it is a Future, so the outer await is sufficient.

**Review by SandPod — APPROVED** [review]
LGTM! 🦖 
Good catch! 👍

**Pull request merged**